### PR TITLE
Fix/airflow settings update

### PR DESCRIPTION
**Pull request by adam2k**
## Description

Use the correct `astro dev start` commands for Airflow 1.0 and Airflow 2.0

## 🎟 Issue(s)

Resolves astronomer/issues#1985

## 🧪 Functional Testing

1. Initialize new project `astro dev init`
2. Update the Dockerfile with `FROM quay.io/astronomer/ap-airflow-dev:2.0.0-1.beta3-buster-onbuild`
3. Add a connection and variable in airflow_settings.yaml, for example:
```
  connections:
    - conn_id: 'local_postgres'
      conn_type: 'postgres'
      conn_host: 'postgres'
      conn_schema: 'postgres'
      conn_login: 'postgres'
      conn_password: 'postgres'
      conn_port: 5432
      conn_extra:
variables:
    - variable_name: 'email_list'
      variable_value: 'adam@astronomer.io'
```
4. `astro dev start` and navigate to the Admin Variables and make sure the variables in the `airflow_settings.yaml` are added to the environment.

## 📸 Screenshots

Screen Shot 2020-11-20 at 11 40 26 AM

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
@adam2k please add tests.

**Comment by adam2k** [comment]
> @adam2k please add tests.

There are already tests for `newAirflowStartCmd`, but I'll see if there is a good way to prevent mocking out of the methods that I changed.

File: airflow/docker.go [hunk 1]
```diff
@@ -211,6 +212,22 @@ func Start(airflowHome string, envFile string) error {
...
+
+	_, airflowTag := docker.GetImageTagFromParsedFile(cmd)
+
+	semVer, err := semver.NewVersion(airflowTag)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse dockerfile Airflow tag: %s", airflowTag)
+	}
+
+	airflowDockerVersion := semVer.Major()
```
Comment (andriisoldatenko) [inline airflow/docker.go:229]: @adam2k let's create new function `airflowVersionFromDockerFile`

Comment (andriisoldatenko) [reply airflow/docker.go:229]: `airflowDockerVersion = airflowVersionFromDockerFile(dockerfile)`

Comment (adam2k) [reply airflow/docker.go:229]: Do you think this should go in the `airflow/docker.go` file or in the `airflow/airflow.go` file?  I could see it in either place.

Comment (andriisoldatenko) [reply airflow/docker.go:229]: just create private function somewhere around `airflow/docker.go`.

Comment (adam2k) [reply airflow/docker.go]: added and updated the tests.

**Review by andriisoldatenko — APPROVED** [review]

**Comment by schnie** [comment]
Hmm, what happens if the customer isn't using our base image directly, but using some other image instead?

**Comment by andriisoldatenko** [comment]
@adam2k yeah good call from @schnie 
can we add hotfix to make it ignore error? or add warning?

**Comment by schnie** [comment]
Do we have the ability to look at the labels on the image before running?

**Comment by adam2k** [comment]
> Do we have the ability to look at the labels on the image before running?

Maybe with Docker inspect?

**Comment by andriisoldatenko** [comment]
best we can do run some script inside entrypoint or similar.

**Pull request merged**